src/components/Images.vue

```diff
@@ -271,57 +279,162 @@ export default {
     },
     doPush(obj) {
       this.currentCommand = `push ${ obj.imageName }:${ obj.tag }`;
+      this.mainWindowScroll = this.$refs.fullWindow.parentElement.parentElement.scrollTop;
       this.startRunningCommand('push');
       ipcRenderer.send('do-image-push', obj.imageName.trim(), obj.imageID.trim(), obj.tag.trim());
     },
     doBuildAnImage() {
-      this.currentCommand = `build ${ this.imageToBuild }`;
+      const imageName = this.imageToBuild.trim();
+
+      this.currentCommand = `build ${ imageName }`;
       this.fieldToClear = 'imageToBuild';
+      this.postCloseOutputWindowHandler = () => this.scrollToImageOnSuccess(imageName);
       this.startRunningCommand('build');
-      ipcRenderer.send('do-image-build', this.imageToBuild.trim());
+      ipcRenderer.send('do-image-build', imageName);
     },
     doPullAnImage() {
-      this.currentCommand = `pull ${ this.imageToPull }`;
+      const imageName = this.imageToPull.trim();
+
+      this.currentCommand = `pull ${ imageName }`;
       this.fieldToClear = 'imageToPull';
+      this.postCloseOutputWindowHandler = () => this.scrollToImageOnSuccess(imageName);
       this.startRunningCommand('pull');
-      ipcRenderer.send('do-image-pull', this.imageToPull.trim());
+      ipcRenderer.send('do-image-pull', imageName);
+    },
+    /**
+     * syntax of a fully qualified tag could start with <hostname>:<port>/
+     * so a colon precedes a tag only if its followed only by valid tag characters
+     * @param fullImageName {string}
+     * @returns {[string, string]}
+     */
+    parseFullImageName(fullImageName) {
+      const m = /^(.+?):([-._A-Za-z0-9]+)$/.exec(fullImageName);
+
+      return m ? [m[1], m[2]] : [fullImageName, 'latest'];
+    },
+    getImageByNameAndTag(imageName, tag) {
+      return this.images.find(image => image.imageName === imageName && image.tag === tag);
+    },
+    scrollToImage(image) {
+      const row = this.$refs.imagesTable.$el.querySelector(`tr[data-node-id="${ image.imageID }"]`);
+
+      if (row) {
+        this.$nextTick(() => {
+          row.scrollIntoView();
+          row.addEventListener('animationend', this.animationEndHandler);
+          row.classList.add('highlightFade');
+        });
+      } else {
+        console.log(`Can't find row for ${ image.imageName }:${ image.tag } in the image table`);
+      }
     },
+    animationEndHandler(event) {
+      const row = event.target;
+
+      row.classList.remove('highlightFade');
+      row.removeEventListener('animationend', this.animationEndHandler);
+    },
+    /**
+     * Does three things:
+     * 1. Verifies the operation ran successfully - in which case there might be a new image
+     * 2. If successful, finds the image in the table
+     * 3. Scrolls to that image and highlights it (via `scrollToImage()`)
+     */
+    scrollToImageOnSuccess(taggedImageName) {
+      if (this.imageManagerOutput.trimStart().startsWith('Error:')) {
+        this.imageManagerOutput = '';
```

Comment:
Are we wiping out the error on failure (so the user can't seem them)? That doesn't seem great.

Reply:
No, this handler fires either when there's no output to display, or the user has dismissed the output-window field.

Reply:
Ah, this was called as `postCloseOutputWindowHandler`… a comment here about that might be useful (because the function name itself seems to imply that this may also be called on non-success)